src/transformers/models/wav2vec2_with_lm/processing_wav2vec2_with_lm.py

```diff
@@ -340,9 +360,61 @@ def batch_decode(
                 </Tip>
 
         Returns:
-            [`~models.wav2vec2.Wav2Vec2DecoderWithLMOutput`] or `tuple`.
+            [`~models.wav2vec2.Wav2Vec2DecoderWithLMOutput`].
+
+        Example:
+
+        ```python
```

Comment:
This example is a bit too long for a docstring IMO, can you mvoe it to the doc page instead? (in docs/source/em/models/wav2vec2_with_lm.mdx)

Reply:
I'm not sure if I understood your suggestion @sgugger. `Wav2Vec2ProcessorWithLM` seems to be documented within [docs/source/en/model_doc/wav2vec2.mdx](https://github.com/huggingface/transformers/blob/d92e22d1f28324f513f3080e5c47c071a3916721/docs/source/en/model_doc/wav2vec2.mdx#wav2vec2processorwithlm). Should I add the example as a custom section (after Overview?), similarly to [Speech2Text#Inference](https://github.com/huggingface/transformers/blob/d92e22d1f28324f513f3080e5c47c071a3916721/docs/source/en/model_doc/speech_to_text.mdx#inference)?

Reply:
You can add the example after the documentation of the processor class.

Reply:
Sorry to bother you, @sgugger, but I still can't grasp how to proceed. I've tried things like below, but `doc-builder` renders the `<div>`s incorrectly.
````markdown
[[autodoc]] Wav2Vec2ProcessorWithLM
    ...
    - batch_decode
Example:

```python
>>> ...
```
    - decode
````

Reply:
The example should be after the whole autodoc block. You can introduce it with a small sentence.

Reply:
Thanks for the explanation @sgugger. I've refactor the example. FYI, I added a link from `batch_decode` to the example section (see [here](https://github.com/huggingface/transformers/blob/ee2f3b81e0af9f770dd7013669dd824b8d522137/src/transformers/models/wav2vec2_with_lm/processing_wav2vec2_with_lm.py#L366)). I hope it's ok to have this cross-reference (`.py`'s docstring referencing a section from `.mdx`).

Now I've also realized that [`[~model.wav2vec2...]`](https://moon-ci-docs.huggingface.co/docs/transformers/pr_18351/en/model_doc/wav2vec2#decoding-multiple-audios) were not rendered properly. Are they available only inside an `autodoc` context?

Reply:
See my comment below, the links are not right (you don't need the full path but you do need the class name!)